### DISCUSSION TITLE
Direct local .aar file dependencies are not supported when building an AAR.

### DISCUSSION THREAD

**Discussion by wilsolutions** [callstack/react-native-builder-bob#346] [Q&A]
Hi, please, I've create a react native module that includes 2 external libraries (.aar).

It builds fine, but when I ran `npx react-native run-android example --deviceId=the_device` it returns the following error:

`Direct local .aar file dependencies are not supported when building an AAR. The resulting AAR would be broken because the classes and Android resources from any local .aar file dependencies would not be packaged in the resulting AAR. Previous versions of the Android Gradle Plugin produce broken AARs in this case too (despite not throwing this error).`

I'm including the .aar in the **build.gradle**:
`implementation files('lovely/cutelib.aar')`

And then in the **settings.gradle** i do:
`include ':cutelib'`

Note: it will run fine into Android device if I don't pass the --deviceId flag :P

Any thoughts?
ty

**Comment by florindum** [comment]
@wilsolutions did you find any solution?

**Comment by jiroscripts** [comment]
Hello @wilsolutions, @florindumitru  

Are you using "aar" in package.json > react-native-builder-bob > targets ?

I'm actually able to embed 3 aar by removing this target option.

My aar(s) are inside a `libs` folder into the `android` folder

Then in my `android/build.gradle` I have:

```
dependencies {
  implementation fileTree(dir: "libs", include: ["*.aar"])
}
```

Also my Android Gradle Plugin Version is in 4.2.1:
(I don't know if it makes any difference)

```
  dependencies {
    classpath 'com.android.tools.build:gradle:4.2.1'
  }
```

**Comment by abraini-Nascent** [comment]
Hello, the issue is still present even when setting the gradle version to 4.2.1.

It will work for debug builds however release builds of apps using RN > 0.64 that use our modules will fail with:
```
[RUN_GRADLEW] Execution failed for task ':rnModule:bundleReleaseLocalLintAar'.
[RUN_GRADLEW] > Error while evaluating property 'hasLocalAarDeps' of task ':rnModule:bundleReleaseLocalLintAar'
```
with the same description @wilsolutions posted.

Does anyone have a solution that will work for the new React Native versions?

**Comment by a-eid** [comment]
did you guys find a workaround ?

**Comment by cristiano-linvix** [comment]
> Hello, the issue is still present even when setting the gradle version to 4.2.1.
> 
> It will work for debug builds however release builds of apps using RN > 0.64 that use our modules will fail with:
> 
> ```
> [RUN_GRADLEW] Execution failed for task ':rnModule:bundleReleaseLocalLintAar'.
> [RUN_GRADLEW] > Error while evaluating property 'hasLocalAarDeps' of task ':rnModule:bundleReleaseLocalLintAar'
> ```
> 
> with the same description @wilsolutions posted.
> 
> Does anyone have a solution that will work for the new React Native versions?

Any solution?

**Comment by DaisukiTamago** [comment]
this stackoverflow answer solved the problem for me: https://stackoverflow.com/a/73699520/10933691

**Comment by Dhruv2110** [comment]
Found the solution. Instead of `./gradlew assembleRelease` try `./gradlew app:assembleRelease`. Now APK build is working fine even with `./gradlew assembleRelease`.

**Comment by Noitham** [comment]
Feels like this is still an unanswered issue, is there any workaround / solution around it?
CC: @atlj

**Comment by faywong** [comment]
You can try to upload the local aar into a remote maven repo to solve this problem.

**Reply by shimulxx** [reply]
How to upload ?

**Comment by Noitham** [comment]
Any solution as of now?

**Reply by DanielAraldi** [reply]
I have the same problem. I tried without `implementation` and used `compileOnly`, build works! But, my native module exists when it's imported 🥲
